src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/CoverageVisualisationService.kt

```diff
@@ -71,15 +75,37 @@ class CoverageVisualisationService(private val project: Project) {
             relativeMutations = (coveredMutations.toDouble() / allMutations * 100).roundToInt()
         }
 
-        visualisationService.panelTitleAbsolute.text = "Absolute Test Coverage for ${testReport.UUT}"
-        visualisationService.panelTitleRelative.text = "Relative Test Coverage for ${testReport.UUT}"
+        // Change the values in the table
+        visualisationService.data[4] = testReport.UUT
+        visualisationService.data[5] = "$relativeLines% ($coveredLines/$allLines)"
+        visualisationService.data[6] = "$relativeBranch% ($coveredBranches/$allBranches)"
+        visualisationService.data[7] = "$relativeMutations% ($coveredMutations/ $allMutations)"
+    }
+
+    /**
+     * Creates a new toolWindow tab for the coverage visualisation.
+     */
+    private fun createToolWindowTab() {
+        val visualisationService = project.service<CoverageToolWindowDisplayService>()
+
+        // Remove coverage visualisation from content manager if necessary
+        val toolWindowManager = ToolWindowManager.getInstance(project).getToolWindow("TestGenie")
+        val contentManager = toolWindowManager!!.contentManager
+        for (content in contentManager.contents) {
+            if (content.displayName.equals("Coverage Visualisation")) {
+                contentManager.removeContent(content, true)
+            }
+        }
 
-        visualisationService.absoluteLines.text = "Amount of Lines covered: $coveredLines Total: $allLines"
-        visualisationService.absoluteBranch.text = "Amount of Branches covered: $coveredBranches Total: $allBranches"
-        visualisationService.absoluteMutant.text = "Amount of Mutants covered: $coveredMutations Total: $allMutations"
+        // If there is no coverage visualisation tab, make it
+        val contentFactory: ContentFactory = ContentFactory.SERVICE.getInstance()
+        val content = contentFactory.createContent(
+            visualisationService.mainPanel, "Coverage Visualisation", true
+        )
+        contentManager.addContent(content)
 
-        visualisationService.relativeLines.text = "Percentage of Lines covered: $relativeLines%"
-        visualisationService.relativeBranch.text = "Percentage of Branches covered: $relativeBranch%"
-        visualisationService.relativeMutant.text = "Percentage of Mutants covered: $relativeMutations%"
+        // Focus on coverage tab and open toolWindow if not opened already
+        contentManager.setSelectedContent(content)
+        toolWindowManager.show()
```

Comment:
That's very nice, I didn't know you could do that so easily!

Reply:
Thanks :)